File: student.api/Model/student.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace student.api.Model
{
    [Table("studentMaster")]
    public class Student
    {
        public int studentId { get; set; }
        [Required]
        public string studName { get; set; } = string.Empty;
        [Required , MaxLength(15 )]
        public string mobileNo { get; set; } = string.Empty;
        [Required]

        public string email { get; set; } = string.Empty;
        public string city { get; set; } = string.Empty;
        public string state { get; set; } = string.Empty;
        public string pincode { get; set; } = string.Empty;
        public string adressline1 { get; set; } = string.Empty;
        public string addressline2 { get; set; } = string.Empty; 

    }
}

File: StudentApp/src/app/students/student/student.ts
import { Component, inject, Input } from '@angular/core';
import { type studentModel } from './student.Model';
import { StudentService } from '../../../services/student.service';

@Component({
  selector: 'tr[app-student]',
  styleUrl: './student.css',
  templateUrl: './student.html',
})
export class Student {
  @Input({ required: true }) student: studentModel = {
    studentId: 0,
    studName: '',
    mobileNo: '',
    email: '',
    city: '',
    state: '',
    pincode: '',
    adressline1: '',
    addressline2: '',
  };
  @Input() index: number = 0;
  
  private studentService = inject(StudentService);


  onEdit() {
    // Pass a copy of the student object to the form so typing doesn't mutate the table row instantly
    this.studentService.setStudentForEdit({ ...this.student });
  }


  onDelete() {
    // Native confirmation dialog matching quick CRUD flows
    if (confirm(`Are you sure you want to delete ${this.student.studName}?`)) {
      this.studentService.DeleteStudent(this.student.studentId);
    }
  }
}